src/app/clusters/service-area-server/service-area-server.h

```diff
@@ -0,0 +1,450 @@
+    BitMask<ServiceArea::Feature> mFeature;
+
+    //*************************************************************************
+    // core functions
+
+    /**
+     * @brief Read Attribute - inherited from AttributeAccessInterface
+     * @return appropriately mapped CHIP_ERROR if applicable
+     */
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+    /**
+     * @brief Command handler - inherited from CommandHandlerInterface
+     * @param[in, out] ctx command context
+     */
+    void InvokeCommand(HandlerContext & ctx) override;
+
+    //*************************************************************************
+    // attribute readers
+
+    /**
+     * @brief Read the Supported Locations Attribute
+     * @param [in, out] aEncoder attribute encoder.
+     * @return appropriately mapped CHIP_ERROR if applicable
+     */
+    CHIP_ERROR ReadSupportedLocations(chip::app::AttributeValueEncoder & aEncoder);
+
+    /**
+     * @brief Read the Supported Maps Attribute
+     * @param [in, out] aEncoder attribute encoder.
+     * @return appropriately mapped CHIP_ERROR if applicable
+     */
+    CHIP_ERROR ReadSupportedMaps(chip::app::AttributeValueEncoder & aEncoder);
+
+    /**
+     * @brief Read the Selected Locations Attribute
+     * @param [in, out] aEncoder attribute encoder.
+     * @return appropriately mapped CHIP_ERROR if applicable
+     */
+    CHIP_ERROR ReadSelectedLocations(chip::app::AttributeValueEncoder & aEncoder);
+
+    /**
+     * Read the Progress Attribute
+     * @param [in, out] aEncoder attribute encoder.
+     * @return appropriately mapped CHIP_ERROR if applicable
+     */
+    CHIP_ERROR ReadProgress(chip::app::AttributeValueEncoder & aEncoder);
+
+    //*************************************************************************
+    // command handlers
+
+    /**
+     * @brief Handle Command: SelectLocations.
+     * @param[in, out] ctx Returns the Interaction Model status code which was user determined in the business logic.
+     *                     If the input value is invalid, returns the Interaction Model status code of INVALID_COMMAND.
+     * @param[in] req the command parameters
+     */
+    void HandleSelectLocationsCmd(HandlerContext & ctx, const Commands::SelectLocations::DecodableType & req);
+
+    /**
+     * @brief Handle Command: SkipCurrentLocation.
+     * @param[in, out] ctx Returns the Interaction Model status code which was user determined in the business logic.
+     *                     If the input value is invalid, returns the Interaction Model status code of INVALID_COMMAND.
+     */
+    void HandleSkipCurrentLocationCmd(HandlerContext & ctx);
+
+    //*************************************************************************
+    // attribute notifications
+
+    /**
+     * @brief Notify MATTER that the attribute has changed
```

Comment:
Could we clean up some of these doc-comments? The copy and pasted 3-line of "notify that attribute changed" does not seem to be useful and overall comments seem to re-state code like `HandleSkipCurrentLocation` says `Handle Command: SkipCurrentLocation` or similar.

Reply:
I have revisited all of the code documentation in the Service Area cluster and remove unnecessary docs and improved others in [this commit](https://github.com/project-chip/connectedhomeip/pull/33991/commits/f1bb1166abfb7871ded52e577b484ace1c7a45be).